### PR TITLE
Update MongoDB week 3

### PR DESCRIPTION
**Pull request by mickyginger**
## What does this change?

Module: MongoDB
Week(s): 3

## Description

During the last iteration in London we opted to deviate from the syllabus as we
decided that it was too advanced for the students. This was what we came up
with instead, and feel it's a better lesson overall given the amount of
experience of MongoDB that students will have at this point in the course.

We appreciate this does not fall into the same format as the rest of the
syllabus, however we feel it is a good starting point to work from.

## Who needs to know about this?

@ChrisOwen101

### CODE REVIEW

File: docs/mongodb/week-3/lesson.md [hunk 1]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
 sidebar_label: Lesson
 ---
 
-Outline:
+## Outline
```
Comment (nbogie) [inline docs/mongodb/week-3/lesson.md:7]: .

Comment (nbogie) [reply docs/mongodb/week-3/lesson.md:7]: Title needs changed from "Advanced Mongo"  (viewed locally at http://localhost:3000/mongodb/index )

 image

(Hi!)

File: docs/mongodb/week-3/lesson.md [hunk 2]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
...
-- Connecting to a MongoDB server from a local Node server
-- Creating and dropping databases and collections in Atlas
-- Reading documentation
-- BSON types
-- Creating documents
-- Creating a document in Atlas
-- Creating a document with the MongoDB Node Driver
-- Updating a document
-- Deleting a document
+Learners should have already created an [Atlas account](https://www.mongodb.com/cloud/atlas), have [Compass](https://www.mongodb.com/products/compass) installed. If not you will need to allow time for them to get set up.
```
Comment (nbogie) [inline docs/mongodb/week-3/lesson.md:42]: ## Suggest bullet pts for second 2nd para clarity:

Learners:
* should create a database for this session.
* should have a database user that can connect to this database. 
* should be able to connect to this database with Compass.

File: docs/mongodb/week-3/lesson.md [hunk 3]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
...
 
-Now that we know how to read databases, collections, and documents, we are ready to move on to more advanced tasks.
+Learners should create a database for this session, they should have a database user that can connect to this database. Students should be able to connect to this database with Compass.
 
-### Use the sample database in MongoDB Atlas
```
Comment (nbogie) [inline docs/mongodb/week-3/lesson.md:41]: ## Suggest: Give ONE course of action

This section is a bit woolly and offers too much choice.  Better to dictate to teachers and TAs what to do.

I suggest we include a short exercise that takes everyone through doing exactly the same db setup, user setup, data load (if necessary).  (or point at one from a previous week, parameterised with a different db name).

File: docs/mongodb/week-3/lesson.md [hunk 4]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
...
+Imagine an app (Netflix, Amazon) is showing you a list of things (movies, products) etc.
 
-Transfer 40 monads from account 1 to account 2. Observe how it works perfectly fine.
+Trace the journey of the request:
 
-## MongoDB Project
+BROWSER -> REACT -> HTTP -> SERVER -> MONGODB -> SERVER -> HTTP -> REACT -> BROWSER
 
-Based on the data from the AirBnB listings and reviews collection, we want to create a front end for it.
+Learners can use any method they like to describe each stage, it could be a diagram or code snippets if they prefer.
```
Comment (nbogie) [inline docs/mongodb/week-3/lesson.md]: A couple of possible presentations of such flows:
[Full-stack CRUD flow - Neill.pdf](https://github.com/CodeYourFuture/syllabus/files/5364632/Full-stack.CRUD.flow.-.Neill.pdf) (available as google sheet at https://docs.google.com/spreadsheets/d/1yxMpMPd0f9T_c1s2h9KRQ_DnYVJ6Vym_23bRS7ZeBRg/edit#gid=0 - but I appreciate we'd want to put it in markdown)


And this one from @DomVinyard 
image

Comment (nbogie) [reply docs/mongodb/week-3/lesson.md]: oops i guess these should go in the corresponding instructor notes document.

File: docs/mongodb/week-3/lesson.md [hunk 5]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
...
+    "Gustav Frèhlich",
+    "Rudolf Klein-Rogge",
+    "Fritz Rasp"
+  ],
+  "directors": [
+    "Fritz Lang"
+  ],
+  "runtime": 153
+}
+```
```
Comment (nbogie) [inline docs/mongodb/week-3/lesson.md]: Add Mike's pseudo-code solution in instructor notes for teachers to crib from.

**Review by nbogie — COMMENTED** [review]
Nice!  The shape is good - would need fleshed out for other teachers to be able to deliver based off this, but I think this is just supposed to be a draft.

File: docs/mongodb/week-3/lesson.md [hunk 6]
```diff
@@ -4,309 +4,142 @@ title: MongoDB - 3
...
-- Role based authentication
-  - cluster based roles
-  - backup and restoration roles
-- Aggregation
-- Replication
-- Transactions
-- A project in MongoDB
-- Next steps
+- This session is all about recapping the previous two weeks of MongoDB and also the Node.js API concepts.
+- The idea is to have a very hands on session that allows the learners to cement their undertsanding.
```
Comment (alismith-dev) [inline docs/mongodb/week-3/lesson.md]:
```suggestion
- The idea is to have a very hands on session that allows the learners to cement their understanding.
```

**Comment by 40thieves** [comment]
Given the beta nature, would it be a good to add a note about this to the instructor/mentor notes?

**Comment by ChrisOwen101** [comment]
@all-contributors add @mickyginger for content

**Review by ChrisOwen101 — APPROVED** [review]

**Pull request closed without merging**